### PR TITLE
Reset file object position in is_png()

### PR DESCRIPTION
**Pull request by chunan**
Otherwise chunk() function will miss the first 8 bytes when `png` is file-object-like.

### CODE REVIEW

File: apng/__init__.py [hunk 1]
```diff
@@ -30,12 +30,14 @@ def is_png(png):
...
+			png_header = f.read(8)
 			
 	if hasattr(png, "read"):
-		png = png.read(8)
+		position = png.tell()
+		png_header = png.read(8)
+		png.seek(position)
 		
-	return png[:8] == PNG_SIGN
+	return png_header[:8] == PNG_SIGN
```
Comment (eight04) [inline apng/__init__.py]: Good catch. However bytes object is not handled correctly. Please use something like:
```py
if isinstance(png, str):
	...
elif hasattr(png, "read"):
	...
elif isinstance(png, bytes):
	png_header = png[:8]
else:
	raise TypeError("Muse be file, bytes, or str")
```
Also no need to slice it again:
```py
return png_header == PNG_SIGN
```

**Review by eight04 — CHANGES_REQUESTED** [review]

File: apng/__init__.py (hunk 1 above)
Comment (chunan) [reply apng/__init__.py]: Done.

**Comment by chunan** [comment]
Updated as requested. PTAL.

**Comment by eight04** [comment]
👍 I'll write some tests before releasing it.

**Review by eight04 — APPROVED** [review]

**Pull request merged**